src/MembershipManager.php

```diff
@@ -133,10 +133,35 @@ public function getMembership(EntityInterface $group, $user_id, array $states =
       $user_id = $user_id->id();
     }
 
-    foreach ($this->getMemberships($user_id, $states) as $membership) {
-      if ($membership->getGroupEntityType() === $group->getEntityTypeId() && $membership->getGroupId() === $group->id()) {
-        return $membership;
-      }
+    // When an empty array is passed, retrieve memberships with all possible
+    // states.
+    $states = $this->prepareConditionArray($states, OgMembership::ALL_STATES);
+
+    $cid = [
+      __METHOD__,
+      $group->getEntityTypeId(),
+      $group->id(),
+      $user_id,
+      implode('|', $states),
+    ];
+    $cid = implode(':', $cid);
+
+    // Use cached result if it exists.
+    if (!$membership_ids = $this->staticCache->get($cid)->data ?? []) {
```

Comment:
Slightly more readable

```suggestion
    // Use cached result if it exists.
    $membership_ids = $this->staticCache->get($cid)->data ?? []
    if (!$membership_ids) {
```